### PR TITLE
Password check

### PR DESCRIPTION
**Pull request by bartosz-gruszczyk**
My solution for password-check with bouns task.

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @bartosz-gruszczyk

**Comment by ziobron** [comment]
⏱ 3 XP extra for @bartosz-gruszczyk for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @bartosz-gruszczyk

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,52 @@
...
+        return "Password needs to have at least one uppercase letter";
+    }
+    if (error == ErrorCode::PasswordsDoNotMatch) {
+        return "Passwords do not match";
+    }
+    return "Ok";
+}
+
+bool doPasswordsMatch(const std::string& firstPassword, const std::string& secondPassword) {
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:25]: Redundant new line.

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,52 @@
+#include <algorithm>
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+std::string getErrorMessage(const ErrorCode& error) {
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:5]: Redundant new line.

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,52 @@
...
+    }
+    if (error == ErrorCode::PasswordsDoNotMatch) {
+        return "Passwords do not match";
+    }
+    return "Ok";
+}
+
+bool doPasswordsMatch(const std::string& firstPassword, const std::string& secondPassword) {
+
+    return (firstPassword == secondPassword);
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:26]: `()` are redundant

File: homework/password-check/validation.cpp [hunk 4]
```diff
@@ -1,2 +1,52 @@
...
+}
+
+ErrorCode checkPasswordRules(const std::string& password) {
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:30]: Redundant new line.

File: homework/password-check/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,52 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    if(std::all_of(password.cbegin(), password.cend(), isalnum)) {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    }
+    return ErrorCode::Ok;
+}
+
+ErrorCode checkPassword(const std::string& firstPassword, const std::string& secondPassword) {
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:47]: Redundant new line.

File: homework/password-check/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,52 @@
...
+}
+
+bool doPasswordsMatch(const std::string& firstPassword, const std::string& secondPassword) {
+
+    return (firstPassword == secondPassword);
+}
+
+ErrorCode checkPasswordRules(const std::string& password) {
+
+    if(password.size() < 9) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:31]: Add space after `if`, `for`, `while` etc...

**Review by nauka-programowania-MA — APPROVED** [review]

File: homework/password-check/validation.cpp [hunk 7]
```diff
@@ -1,2 +1,52 @@
...
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    }
+    if(!std::any_of(password.cbegin(), password.cend(), isupper)) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:34]: Better us `none_of`

File: homework/password-check/validation.cpp [hunk 8]
```diff
@@ -1,2 +1,52 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    }
+    if(!std::any_of(password.cbegin(), password.cend(), isdigit)) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:37]: Better us none_of

File: homework/password-check/validation.cpp [hunk 9]
```diff
@@ -1,2 +1,52 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    if(std::all_of(password.cbegin(), password.cend(), isalnum)) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:40]: Better us none_of

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**